x-pack/solutions/security/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/upgrade_after_preview.md

```diff
@@ -0,0 +1,681 @@
+
+## Useful information
+
+### Tickets
+
+- [Rule Immutability/Customization](https://github.com/elastic/security-team/issues/1974) epic (internal)
+
+**Milestone 3 - Prebuilt Rules Customization:**
+
+- [Milestone 3 epic ticket](https://github.com/elastic/kibana/issues/174168)
+- [Tests for prebuilt rule upgrade workflow #202078](https://github.com/elastic/kibana/issues/202078)
+
+### Terminology
+
+- **CTA**: "call to action", usually a button, a link, or a callout message with a button, etc, that invites the user to do some action.
+
+  - **CTA to upgrade the prebuilt rule** - a button to upgrade the current prebuilt rule shown in a flyout
+
+- **field is non-customized**: rule's field has an original value obtained after prebuilt rule installation
+
+- **field is customized**: rule's field has a value semantically different from ann original one obtained after prebuilt rule installation
+
+- **rule is installed**: rule has been installed and doesn't have any customized fields
+
+- **rule is non-customized**: rule doesn't have any customized fields
+
+- **rule is customized**: rule has one or more customized fields
+
+### Assumptions
+
+- Below scenarios only apply to prebuilt detection rules.
+- A rule is shown on Rule Upgrade page when it has an upgrade.
+
+### Non-functional requirements
+
+- Notifications, rule installation and rule upgrade workflows should work:
+  - regardless of the package type: with historical rule versions or without;
+  - regardless of the package registry availability: i.e., they should also work in air-gapped environments.
+- Rule installation and upgrade workflows should work with packages containing up to 15000 historical rule versions. This is the max number of versions of all rules in the package. This limit is enforced by Fleet.
+- Kibana should not crash with Out Of Memory exception during package installation.
+- For test purposes, it should be possible to use detection rules package versions lower than the latest.
+
+### Functional requirements
+
+- User should be able to install prebuilt rules with and without previewing what exactly they would install (rule properties).
+- User should be able to upgrade prebuilt rules with and without previewing what updates they would apply (rule properties of target rule versions).
+- If user chooses to preview a prebuilt rule to be installed/upgraded, we currently show this preview in a flyout.
+- In the prebuilt rule preview a tab that doesn't have any sections should not be displayed and a section that doesn't have any properties also should not be displayed.
+- Rule preview shows only [customizable fields](./shared_assets/customizable_rule_fields.md)
+- User should be able to upgrade a prebuilt rule with upgrades in the [non-customizable fields](./shared_assets/non_customizable_rule_fields.md)
+- Any other rule and alerting framework fields non involved in the rule upgrade workflow like `enabled` or
+  `exceptions` should stay unchanged after rule upgrade.
```

Comment:
This contains some irrelevant information. I'll make it more specific and add more requirements.

Reply:
Addressed in https://github.com/elastic/kibana/pull/205645/commits/77a486a7a56dde7ab921d66ec69fed99b7f30b04